lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, appendData = []) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return Promise.resolve(true)
+      .then(() => {
+        return this.client({
+          method,
+          url,
+          params,
+          data
+        })
+      })
+      .then(response => {
+        if (response.error) throw r.error
+
+        const responseData = Array.isArray(response.data.data)
+          ? [...response.data.data, ...appendData]
+          : response.data.data
+
+        if (
+          method === 'GET' &&
+          response.data.meta.pagination.totalPages >
+            response.data.meta.pagination.page
+        ) {
+          return this.fetch(
+            endpoint,
+            {
+              ...params,
+              page: (params.page || 1) + 1
+            },
+            data,
+            responseData
+          )
+        }
+
+        return responseData
+      })
+  }
+
+  async walletBalance(walletId, index, confirmations = 1) {
+    return this.fetch('transactions', {
+      walletId,
+      accountIndex: index
+    })
+      .then(transactions =>
+        transactions.reduce((acc, item) => {
+          if (item.confirmations < confirmations) {
+            return acc
+          }
+
+          return acc + (item.direction === 'outgoing' ? -1 : 1) * item.amount
+        }, 0)
+      )
+      .then(r => BN(r).round())
+  }
+
+  async addressBalance(address, confirmations) {
+    return this.fetch('transactions', {
+      address: address
+    })
+      .then(transactions =>
+        transactions.reduce((acc, item) => {
+          const inputs = item.inputs
+            .filter(
+              record =>
+                record.address === address &&
+                item.confirmations >= confirmations
+            )
+            .reduce((acc, item) => acc + item.amount, 0)
+
+          const outputs = item.outputs
+            .filter(
+              record =>
```

Comment:
we repeat the same logic here. Maybe we can extract it to a function with some good name?

Reply:
maybe `sumBalanceByAddressAndCofirmations`, but I'm not really sure about that name... if this function will be local, maybe only `sumBatchBalance`?

Reply:
I like the longer one. It is quite long however it would be clear what it does. Or maybe just `sumBalancForAddress(address, minConfirmations)`. So when you rename `confirmations` -> `minConfirmations` also this shorter variant might be clear enough.